### PR TITLE
KIN-71: Adding default strategy.

### PR DESCRIPTION
**Pull request by deandalm**
#### Purpose
- [X] Feature
- [ ] Bug Fix
- [ ] Hotfix
- [ ] Other

#### Associated Tickets
- [Default Strategy](https://silvercar.atlassian.net/browse/KIN-71)

#### Details
- Adding Default Strategy that is always enabled.

#### Checklist
- [X] Tests
- [ ] Correct base and merge branches

### CODE REVIEW

File: unleash/Strategies/DefaultStrategy.swift [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+//  Unleash
+//
+//  Copyright © 2019 Silvercar. All rights reserved.
+//
+
+import Foundation
+
+class DefaultStrategy : Strategy {
+    func getName() -> String {
+        return "default"
```
Comment (deandalm) [inline unleash/Strategies/DefaultStrategy.swift]: Is this the best way to handle a magic string in Swift?

Comment (alamantia) [reply unleash/Strategies/DefaultStrategy.swift]: I used to do something like a constants struct or an extension

Comment (deandalm) [reply unleash/Strategies/DefaultStrategy.swift]: If I created an extension to String in this file, would it only be accessible for this file? I don't want a global constants file.

Comment (alamantia) [reply unleash/Strategies/DefaultStrategy.swift]: I think private and fileprivate work with extensions if that's what you want to do

Comment (deandalm) [reply unleash/Strategies/DefaultStrategy.swift]: @alamantia like this:
```
private extension String {
    static let name: String = "default"
}

class DefaultStrategy : Strategy {
    func getName() -> String {
        return .name
    }
    
    func isEnabled(parameters: [String : String]) -> Bool {
        return true
    }
}
```

Comment (louisvdaily) [reply unleash/Strategies/DefaultStrategy.swift]: @deandalm Yes, like this. However, making a function “getName” isn’t really swift like when you can instead have a computed property that returns the string value.

Comment (deandalm) [reply unleash/Strategies/DefaultStrategy.swift]: @louisvdaily here is the protocol I wrote, how should it be changed:
```
public protocol Strategy {
    func getName() -> String
    func isEnabled(parameters: [String: String]) -> Bool
}
```

Comment (louisvdaily) [reply unleash/Strategies/DefaultStrategy.swift]: @deandalm I’m doing this from my phone, so please excuse me. You can do so like this:

var name: String { get }

In place of:

func getName() -> String

In use:

class DefaultStrategy: Strategy {
  let name: String {
    return “default”
  }

  // extra class stuff here
}

**Pull request merged**